Core/include/Acts/Propagator/ConstrainedStep.hpp

```diff
@@ -30,137 +48,120 @@ struct ConstrainedStep {
   /// from user     - this is user given for what reason ever
   enum Type : int { accuracy = 0, actor = 1, aborter = 2, user = 3 };
 
-  /// the step size tuple
-  std::array<Scalar, 4> values = {
-      {std::numeric_limits<Scalar>::max(), std::numeric_limits<Scalar>::max(),
-       std::numeric_limits<Scalar>::max(), std::numeric_limits<Scalar>::max()}};
+  /// Number of iterations needed by the stepsize finder
+  /// (e.g. Runge-Kutta) of the stepper.
+  size_t nStepTrials = std::numeric_limits<size_t>::max();
 
-  /// The Navigation direction
-  NavigationDirection direction = NavigationDirection::Forward;
+  constexpr ConstrainedStep() = default;
 
-  /// Update the step size of a certain type
+  /// constructor from Scalar
+  /// navigation direction is inferred by the sign of the step size
+  /// @param value is the user given initial value
+  constexpr explicit ConstrainedStep(Scalar value) {
+    m_values[user] = std::abs(value);
```

Comment:
I see, you take the user value as a start, sounds ok.

What if the constrained step is however, created inside a navigator ?

Reply:
The default constructor is defined as well, so

```cpp
ConstrainedStep step{};
```

is valid.

Reply:
this is how it was in the original `ConstrainedStep` https://github.com/acts-project/acts/blob/main/Core/include/Acts/Propagator/ConstrainedStep.hpp#L78

the other step constraints are changed through `setValue` and `update`